samples/kafka-streams-interactive-query/src/main/java/com/example/kafka/streams/music/KafkaStreamsInteractiveQueryApplication.java

```diff
@@ -0,0 +1,183 @@
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonSetter;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Grouped;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.kafka.support.serializer.JsonSerde;
+
+@SpringBootApplication(proxyBeanMethods = false)
+class KafkaStreamsInteractiveQueryApplication {
+
+	private static final Logger LOG = LoggerFactory.getLogger(KafkaStreamsInteractiveQueryApplication.class);
+
+	public static void main(String[] args) {
+		SpringApplication.run(KafkaStreamsInteractiveQueryApplication.class, args);
+	}
+
+	@EnableConfigurationProperties(ProductTrackerProperties.class)
+	static class InteractiveProductCountApplication {
+
+		static final String PRODUCT_TOPIC = "product-events";
+		static final String PRODUCT_COUNT_STORE_NAME = "product-count-store";
+		static final String TOP_TWO_PRODUCTS_STORE_NAME = "top-two-products-store";
+		static final String TOP_TWO_PRODUCTS_KEY = "top-two";
+
+		@Value("${spring.cloud.stream.kafka.streams.binder.configuration.state.dir:UNKNOWN}")
+		private String stateStoreDir;
+
+		private final ProductTrackerProperties productTrackerProperties;
+
+		InteractiveProductCountApplication(ProductTrackerProperties productTrackerProperties) {
+			this.productTrackerProperties = productTrackerProperties;
+		}
+
+		@PostConstruct
+		void showStateStoreInfo() {
+			LOG.info("Using KafkaStreams state.dir = {}", this.stateStoreDir);
+		}
+
+		@Bean
+		public Consumer<KStream<Object, Product>> process() {
+			return (input) -> {
+				// Accept product events that are configured to be tracked
+				final KStream<Integer, Product> productsByProductId = input
+						.filter((key, product) -> productTrackerProperties.getProductIds().contains(product.id()))
+						.map((key, value) -> new KeyValue<>(value.id, value));
+
+				Serde<Product> productSerde = new JsonSerde<>(Product.class);
+				Serde<ProductCount> productCountSerde = new JsonSerde<>(ProductCount.class);
+				Serde<TopTwoProducts> topTwoSerde = new JsonSerde<>(TopTwoProducts.class).noTypeInfo();
```

Comment:
ℹ️ I was able to drop the previously used custom serde impl and instead leverage `JsonSerde` w/ some carefully placed `@JsonIgnore/@JsonGetter/@JsonSetter` on the aggregator `TopTwoProducts` object directly.